Self Hosted GH runner failing to install ArgoCD

I'm using a self hosted GitHub runner in kubernetes (which works for my daily pipelines) and I can't make argocd-diff-preview work with it. I constantly get following errors :

```console
[2025-03-03T14:30:15Z INFO  argocd_diff_preview::parsing] 🤖 Found 11 Application[Sets] before filtering
[2025-03-03T14:30:15Z INFO  argocd_diff_preview::parsing] 🤖 Found 2 Application[Sets] after filtering
[2025-03-03T14:30:15Z INFO  argocd_diff_preview::parsing] 🤖 Patching Application[Sets] for branch: test-preview
[2025-03-03T14:30:15Z INFO  argocd_diff_preview::parsing] 🤖 Patching 2 Argo CD Application[Sets] for branch: test-preview
[2025-03-03T14:30:15Z DEBUG argocd_diff_preview::utils] Running shell command: docker ps
[2025-03-03T14:30:15Z INFO  argocd_diff_preview::kind] 🚀 Creating cluster...
[2025-03-03T14:30:15Z DEBUG argocd_diff_preview::utils] Running shell command: kind delete cluster --name argocd-diff-preview
[2025-03-03T14:30:15Z DEBUG argocd_diff_preview::utils] Running shell command: kind create cluster --name argocd-diff-preview
[2025-03-03T14:30:50Z INFO  argocd_diff_preview::kind] 🚀 Cluster created successfully
[2025-03-03T14:30:50Z DEBUG argocd_diff_preview::utils] Running shell command: kubectl create ns argocd
[2025-03-03T14:30:50Z DEBUG argocd_diff_preview] 🤖 Namespace 'argocd' created successfully
[2025-03-03T14:30:50Z INFO  argocd_diff_preview] 🤷 No secrets found in ./secrets
[2025-03-03T14:30:50Z INFO  argocd_diff_preview::argocd] 🦑 Installing Argo CD Helm Chart version: 'latest'
[2025-03-03T14:30:50Z DEBUG argocd_diff_preview::argocd] 📂 Files in folder 'argocd-config':
[2025-03-03T14:30:50Z DEBUG argocd_diff_preview::argocd] - 📄 "values-override.yaml"
[2025-03-03T14:30:50Z DEBUG argocd_diff_preview::argocd] - 📄 "values.yaml"
[2025-03-03T14:30:50Z DEBUG argocd_diff_preview::utils] Running shell command: helm repo add argo https://argoproj.github.io/argo-helm
[2025-03-03T14:30:51Z DEBUG argocd_diff_preview::utils] Running shell command: helm repo update
[2025-03-03T14:30:51Z DEBUG argocd_diff_preview::utils] Running shell command: helm install argocd argo/argo-cd -n argocd -f argocd-config/values.yaml -f argocd-config/values-override.yaml 
[2025-03-03T14:35:59Z ERROR argocd_diff_preview::argocd] ❌ Failed to install Argo CD
[2025-03-03T14:35:59Z DEBUG argocd_diff_preview::utils] Running shell command: which kind
[2025-03-03T14:35:59Z DEBUG argocd_diff_preview::utils] Running shell command: which kind
[2025-03-03T14:35:59Z ERROR argocd_diff_preview] ❌ Error: INSTALLATION FAILED: failed pre-install: 1 error occurred:
    	* timed out waiting for the condition
    
    
    
[2025-03-03T14:35:59Z DEBUG argocd_diff_preview::utils] Running shell command: kind get clusters
[2025-03-03T14:35:59Z INFO  argocd_diff_preview] 🧼 Cleaning up...
[2025-03-03T14:35:59Z INFO  argocd_diff_preview::kind] 💥 Deleting cluster...
[2025-03-03T14:35:59Z DEBUG argocd_diff_preview::utils] Spawning command: kind delete cluster --name argocd-diff-preview
[2025-03-03T14:36:01Z INFO  argocd_diff_preview::kind] 💥 Cluster deleted successfully
[2025-03-03T14:36:01Z INFO  argocd_diff_preview] If you think this error is caused by a bug, please open an issue on GitHub
Error: CommandError { stderr: "Error: INSTALLATION FAILED: failed pre-install: 1 error occurred:\n\t* timed out waiting for the condition\n\n\n" }
Error: Process completed with exit code 1.
```

I increased resources requests and limits for docker sidecar container to : 
```yaml
      dockerdContainerResources:
        limits:
          cpu: "4"
          memory: "8Gi"
        requests:
          cpu: "500m"
          memory: "1Gi"
```

Any idea on how to troubleshoot this ?

--- a/docs/self-hosted-gh-runner.md
+++ b/docs/self-hosted-gh-runner.md
@@ -1,0 +1,35 @@
+## Self hosted Github Actions Runners on kubernetes
+
+When using Action Runner Controller (ARC) to run self-host your GitHub Actions Runners. You need to ensure that the Service and Pod CIDRs of the kind cluster created by `argocd-diff-preview` don't overlap with your host cluster's CIDRs.
+
+The default CIDRs are:
+
+| Service | CIDR        |
+|---------|-------------|
+| Service | 10.96.0.0/16|
+| Pod     | 10.244.0.0/16|
+
+To configure kind:
+
+1. Create a file in your repo, for instance `hack/kind.yaml`, with the following content:
+```yaml
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+networking:
+  serviceSubnet: "10.80.0.0/16"
+  podSubnet: "10.128.0.0/16"
+```
+2. Add the flag `--kind-options '--config /base-branch/hack/kind.yaml'` to `argocd-diff-preview`.
+```yaml
+      - name: Generate Diff
+        run: |
+          docker run \
+            --network=host \
+            -v /var/run/docker.sock:/var/run/docker.sock \
+            -v $(pwd)/main:/base-branch \
+            -v $(pwd)/pull-request:/target-branch \
+            -v $(pwd)/output:/output \
+            -e TARGET_BRANCH=${{ github.head_ref }} \
+            -e REPO=${{ github.repository }} \
+            dagandersen/argocd-diff-preview:v0.1.8 --kind-options '--config /base-branch/hack/kind.yaml'
+```

